nautobot/dcim/forms.py

```diff
@@ -5157,3 +5176,121 @@ class Meta:
             "weight",
             "tags",
         )
+
+
+#
+# Virtual Device Context
+#
+
+
+class VirtualDeviceContextForm(NautobotModelForm):
+    device = DynamicModelChoiceField(
+        queryset=Device.objects.all(),
+    )
+    tenant = DynamicModelChoiceField(
+        queryset=Tenant.objects.all(),
+        required=False,
+    )
+
+    class Meta:
+        model = VirtualDeviceContext
+        fields = [
+            "name",
+            "device",
+            "status",
+            "identifier",
+            "primary_ip4",
+            "primary_ip6",
+            "tenant",
+            "description",
```

Comment:
I'd expect to be able to assign interfaces from the selected `device` as a part of the VirtualDeviceContext edit form.